### PR TITLE
fix(rest): Added support for pagination and retrival using multi value `projectRelevance` param

### PR DESCRIPTION
**Pull request by smrutis1**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*#1160*)
> * Did you add or update any new dependencies that are required for your change? - No

Issue: CLoses #1160 

Need to test the `projects/{id}/vulnerabilities?projectRelevance=APPLICABLE&projectRelevance=IRRELEVANT` endpoint(may be along with other filters) and verify the outputs if it works fine. 
Need to test the pagination as well. 

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

Signed-off-by: Smruti Prakash Sahoo <smruti.sahoo@siemens.com>

### CODE REVIEW

**Review by JaideepPalit — CHANGES_REQUESTED** [review]
Kindly look into some minor review comments
Tested the PR on stage.
The REST API, fetches data properly for single/multi-value project-relevance and priority

However seems to be some issue with Pagination and sort.
1.	Response contains noVulnerability data for pagination
2.	totalPages and totalElements always shows 20 record. It should be no. of records found based on params
3.	Also tried sort without pagination, doesnot return sorted result

Shared more detailed info in mail.

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java [hunk 1]
```diff
@@ -878,15 +878,23 @@ public void should_document_get_project_vulnerabilities() throws Exception {
...
+                .param("page_entries", "5")
+                .param("sort", "externalId,desc")
                 .accept(MediaTypes.HAL_JSON))
                 .andExpect(status().isOk())
                 .andDo(this.documentationHandler.document(
                         requestParameters(
                                 parameterWithName("priority").description("The priority of vulnerability. For example: `1 - critical`, `2 - major`"),
-                                parameterWithName("projectRelevance").description("The relevance of project of the vulnerability, possible values are: " + Arrays.asList(VulnerabilityRatingForProject.values()))
+                                parameterWithName("projectRelevance").description("The relevance of project of the vulnerability, possible values are: " + Arrays.asList(VulnerabilityRatingForProject.values())),
+                                parameterWithName("page").description("Page of releases"),
```
[the remark this answers is not in the document]
Comment (smrutis1) [reply rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java]: updated it to `vulnerabilities`

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java [hunk 2]
```diff
@@ -878,15 +878,23 @@ public void should_document_get_project_vulnerabilities() throws Exception {
...
+                                parameterWithName("page_entries").description("Amount of vulnerability page"),
+                                parameterWithName("sort").description("Defines order of the vulnerability")
```
[the remark this answers is not in the document]
Comment (smrutis1) [reply rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java]: Mentioned the options

**Comment by smrutis1** [comment]
@JaideepPalit , Thanks for review and test. Have fixed the mentioned issue, kindly have a look. 

And as discussed regarding the 3rd point, as the sorting and paging tightly coupled. Need to be discussed whether to include this as part of this PR or not as the same will be applicable for other endpoints where pagination has been applied

**Comment by JaideepPalit** [comment]
Retested the PR.
1. Response contains noVulnerability data for pagination  - `working fine`
2. totalPages and totalElements always shows 20 record. It should be no. of records found based on params - `working fine`
3. Also tried sort without pagination, doesnot return sorted result - `issue persists`
4. sort with pagination - `works fine`

**Review by JaideepPalit — APPROVED** [review]

**Pull request merged**